argus/backend/service/notification_manager.py

```diff
@@ -11,19 +13,53 @@ class NotificationManagerException(Exception):
 
 
 class NotificationManagerService:
+
+    def __init__(self) -> None:
+        self.notification_services = []
+        self.attach_notification_sender(ArgusDBNotificationSaver())
```

Comment:
Better solution would be passing list of notification senders in `NotificationManagerService` constructor, so we could decouple these classes from each other allowing simpler unit tests (by providing fake sender instead ArugsDB/Email senders)

Reply:
it could be better, if NotificationManagerService created once, some kind of singelton. But now it created Each time when need send , read notification, get status

Reply:
Singletons would be dangerous to use in this context, as flask instantiates new context for each request, meaning using a singleton that lives outside this context could lead to bugs related to data leakage. Sending notifications senders in the constructor would indeed be better, you can make constructor instantiate default ones if None was passed instead of a list of senders

Reply:
but does `notification_manager` depends on flask context? Isn't it a separate service that works in background?

Reply:
For now it only runs within flask context, but doesn't depend on it. Background services aren't implemented yet.